api/envoy/extensions/filters/http/lua/v3/lua.proto

```diff
@@ -78,6 +81,12 @@ message Lua {
   //         stat_prefix: bar_script # This emits lua.bar_script.errors etc.
   //
   string stat_prefix = 4;
+
+  // If set to true, the Lua filter will clear the route cache automatically if the request
+  // headers are modified by the Lua script. If set to false, the Lua filter will not clear the
+  // route cache automatically.
+  // Default is true for backward compatibility.
+  google.protobuf.BoolValue clear_route_cache = 5;
```

Comment:
`clear_route_cache_on_header_mutation` maybe? I think it will be clearer if the condition is included in the name.

Similar for the variable name in the PR.

Reply:
This name is used to keep consistance with our `ext_authz`, `jwt_authn`, `ext_proc`, `language`, etc filters. They have same feature and a `clear_route_cache` API field. Maybe the comments could cover the lost the semantics in the name? Then we can ensure our users could have consistant API.

Reply:
Got it. Probably best to leave it consistent with others.